### DISCUSSION TITLE
Getting `CookieJar` from headers in tests

### DISCUSSION THREAD

**Discussion by aalowlevel** [tokio-rs/axum#2020] [Q&A] [answered]
- [x] I have looked for existing issues (including closed) about this

### Version
axum = { version = "0.6.18", features = ["macros", "headers"] }
axum-extra = { version = "0.7.4", features = ["cookie"] }

### Platform
64-bit (Windows 10)

### Description
`from_headers` function leaves `CookieJar` empty in tests. Consider this code:

I tried this code:

```rust
let client = hyper::Client::new();
    let json_bytes = serde_json::to_vec(&json!({}))
        .unwrap();
    let request = Request::builder()
        .uri("/")
        .method("POST")
        .header("Content-Type", "application/json")
        .body(Body::from(json_bytes))
        .unwrap();
    let response = client
        .request(request)
        .await.unwrap();
    let cookies = CookieJar::from_headers(response.headers());
    dbg!(&cookies);
```

I expected to see this happen:

Instead, this happened:

```
&cookies = CookieJar {
    jar: CookieJar {
        original_cookies: {},
        delta_cookies: {},
    },
}
```

**Comment by davidpdrsn** [comment, marked as answer]
`CookieJar` only supports cookies from requests (`Cookie` headers). Not cookies from responses (`Set-Cookie`)
